homeassistant/components/rova/sensor.py

```diff
@@ -96,33 +111,33 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     data_service = RovaData(api)
 
     # Create a new sensor for each garbage type.
-    entities = []
-    for sensor_key in config[CONF_MONITORED_CONDITIONS]:
-        sensor = RovaSensor(platform_name, sensor_key, data_service)
-        entities.append(sensor)
-
+    entities = [
+        RovaSensor(platform_name, description, data_service)
+        for description in SENSOR_TYPES
+        if description.key in config[CONF_MONITORED_CONDITIONS]
+    ]
```

Comment:
Changed how entities are added. Instead of only iteration over the config values, I now iterate over `SENSOR_TYPES` and add sensor if `key` in `monitored_conditions`.

Reply:
I'd keep the descriptions in a dict and keep the old monitored condition keys as keys in that dict.

Reply:
One of the ideas behind the EntityDescription change was to combine `key` and attributes into one to only pass it around. Not sure if duplicating the `key` field would defeat that purpose.

It still needs to be passed to the entity for the api call.

Reply:
My point here is that the monitored conditions key is not the same as the api call key. But we don't need the monitored conditions key in the entity only the api call key.

Reply:
I'm probably missing something here? Isn't the monitored conditions `key` used as well, for the `name` property? Or did you mean I should change the `desc.name` to match the current `key` since the current `name` isn't used?

Reply:
Yes, let's change desc.name to math the monitored conditions key since that's the only place in the entity where that key is used. Then we can use desc.key for the api call instead and we don't need any extra key attributes, like json_key or json_key_required.